cupyx/scipy/sparse/csr.py

```diff
@@ -1014,3 +1074,76 @@ def cupy_binopt_csr_step2(op_name):
         ''',
         name,
     )
+
+
+def csr2dense(a, order):
+    out = cupy.zeros(a.shape, dtype=a.dtype, order=order)
+    m, n = a.shape
+    cupy_csr2dense()(m, n, a.indptr, a.indices, a.data,
+                     (order == 'C'), out)
+    return out
+
+
+@cupy._util.memoize(for_each_device=True)
+def cupy_csr2dense():
+    return cupy.ElementwiseKernel(
+        'int32 M, int32 N, raw I INDPTR, I INDICES, T DATA, bool C_ORDER',
+        'raw T OUT',
+        '''
+        int row = get_row_id(i, 0, M - 1, &(INDPTR[0]));
+        int col = INDICES;
+        if (C_ORDER) {
+            OUT[col + N * row] += DATA;
+        } else {
+            OUT[row + M * col] += DATA;
+        }
+        ''',
+        'cupy_csr2dense',
+        preamble=_GET_ROW_ID_
+    )
+
+
+def dense2csr(a):
+    m, n = a.shape
+    a = cupy.ascontiguousarray(a)
+    indptr = cupy.zeros(m + 1, dtype=numpy.int32)
+    info = cupy.zeros(m * n + 1, dtype=numpy.int32)
+    cupy_dense2csr_step1()(m, n, a, indptr, info)
+    indptr = cupy.cumsum(indptr, dtype=numpy.int32)
+    info = cupy.cumsum(info, dtype=numpy.int32)
+    nnz = int(indptr[-1])
+    indices = cupy.empty(nnz, dtype=numpy.int32)
+    data = cupy.empty(nnz, dtype=a.dtype)
+    cupy_dense2csr_step2()(m, n, a, info, indices, data)
+    return csr_matrix((data, indices, indptr), shape=(m, n))
+
+
+@cupy._util.memoize(for_each_device=True)
+def cupy_dense2csr_step1():
+    return cupy.ElementwiseKernel(
+        'int32 M, int32 N, T A',
+        'raw I INDPTR, raw I INFO',
+        '''
+        int row = i / N;
+        int col = i % N;
+        if (A == static_cast<T>(0)) return;
```

Comment:
The body of `ElementwiseKenel` is expanded as follows.
https://github.com/cupy/cupy/blob/a118905a7976079db8726001a3983d520847cb52/cupy/core/_kernel.pyx#L41-L52
So, please use `continue` instead of `return`.
```suggestion
        if (A == static_cast<T>(0)) continue;
```

Reply:
Following implementation might be better for readability.
```py
if (A != static_cast<T>(0)) {
   atomicAdd(...)
   ...
}
```